### DISCUSSION TITLE
Spread operator not works as expected in all cases (3.0.0-beta-2042)

### DISCUSSION THREAD

**Discussion by berkid89** [sebastienros/jint#1344] [Q&A] [answered]
Hello!

When I using the JS Spread syntax (...) to construct an another object with extra props, its not working as expected:

```csharp
            engine.Execute("test = (x) => { return {...x, 'newProp':'yeee'}; }");
            var result = engine.Invoke("test", "{'name':'John','age':30}");
```

The result contains only the `newProp`:

```
{ 'newProp':'yeee' }
```

but the expected is:

```
{'name':'John', 'age':30, 'newProp':'yeee' }
```

![image](https://user-images.githubusercontent.com/16096587/199699235-5137ce88-4b80-49ce-888c-efc19019f099.png)

Tell me if i'm doing something wrong :)

**Comment by lahma** [comment, marked as answer]
You are invoking function `test` with string parameter that has value `"{'name':'John','age':30}"` so you are trying to spread string. This shows the difference and expected output `var result = engine.Evaluate("test({'name':'John','age':30})")`, here's object parameter instead of string.

**Reply by berkid89** [reply]
True! Thank you @lahma!

Calling Evaluate() with the correct parameter solved the problem.
